File: src/bindings/swig/python/examples/example_kdb.py
import kdb

# open database
with kdb.KDB() as db:
	ks = kdb.KeySet(100)
	# get keys
	db.get(ks, "user:/MyApp")

	# check if key exists
	try:
		key = ks["user:/MyApp/mykey"]
	except KeyError:
		# create a new key + append to keyset
		key = kdb.Key("user:/MyApp/mykey")
		ks.append(key)

	# change keys value
	key.value = "new_value"

	# store new keyset
	db.set(ks, "user:/MyApp")

# database close is implicit

File: src/bindings/swig/lua/examples/example_kdb.lua
require("kdb")

do
	-- open database
	local db = kdb.KDB()
	ks = kdb.KeySet(100)
	-- get keys
	db:get(ks, "user:/MyApp")

	-- check if key exists
	key = ks["user:/MyApp/mykey"]
	if not key then
		-- create a new key + append to keyset
		key = kdb.Key("user:/MyApp/mykey")
		ks:append(key)
	end
	-- change keys value
	key.string = "new_value"

	-- store new keyset
	db:set(ks, "user:/MyApp")

	-- database closes at end of context is implicit
end